### PR TITLE
Implement VALUE

### PR DESCRIPTION
**Pull request by nicolashefti**

### CODE REVIEW

File: test/text.js [hunk 1]
```diff
@@ -242,9 +242,13 @@ describe('Text', function() {
     text.UPPER(true).should.equal(error.value);
   });
 
-  xit('VALUE', function() {
+  it('VALUE', function() {
     text.VALUE('$1,000').should.equal(1000);
```
Comment (pbadenski) [inline test/text.js:246]: This gives `#VALUE!` on my Excel - thoughts?

Comment (nicolashefti) [reply test/text.js:246]: In mine too. But this works `=VALUE("10,12€") => 10,12`. I guess it has something to do with how the currency and language are set.

I think because of ` num = num.replace(/[^0-9eE\.-]+/g, '');` this version is more permissive than the Excel one: `=formulajs.VALUE("WHATERVER102") => 102`.

Sometimes the spec are not so clear when you guess them from Excel/LibreOffice...

Comment (pbadenski) [reply test/text.js:246]: From what I've found - it will work if in Excel you type number and use "Currency" formatting to format it. It won't work if you use the same contents, but use "General" formatting. In principle Excel removes the formatting which is tricky. If I was implementing this feature I would implement it on framework level - with a data structure that separates raw number and formatting, and VALUE function would just return a raw number.

As you suggested at the moment this version is more permissive (it will even process text like "100+2") - but also won't work for many cases, eg. "123.00 ل.س." or "B/.123.00". As far as I understand ISO 4217 provides detailed spec on this (https://www.foreignexchangelive.com/currency-codes-symbols/)

File: test/text.js [hunk 2]
```diff
@@ -242,9 +242,13 @@ describe('Text', function() {
     text.UPPER(true).should.equal(error.value);
   });
 
-  xit('VALUE', function() {
+  it('VALUE', function() {
```
[the remark this answers is not in the document]
Comment (nicolashefti) [reply test/text.js:304]: Done

File: test/text.js (hunk 1 above)
Comment (nicolashefti) [reply test/text.js:305]: I am wondering what is the best solution: A) implementing (even partially) a function or B) sticking to Excel behavior.

In the A) approach, the current implementation is in a good enough state. The work overload to reach B) seems to be out of the scope of what I had in my head when I started to implement this parsing function. If some developers need a more accurate version, they can propose a new increment with more tests and code.

What do you think?

Comment (pbadenski) [reply test/text.js:305]: That's fair - I believe excel doesn't implement VALUE in the same way. Formatting is probably applied on top of a raw representation which could be both stored and easily accessed. In a way what we're trying to pull off is quite ambitious :D

I think it's a good pragmatic decision to have this minimal VALUE implementation released with maybe some comment for the user. Also - what do you think of adding a special section for some of the "incorrect" test cases we identified as [characterization tests](https://en.wikipedia.org/wiki/Characterization_test) - with a clear note specifying these do not constitute expected behaviour, but document our awareness of it.

Comment (nicolashefti) [reply test/text.js:305]: I like the idea of characterization tests. I will update the code in this direction.

**Comment by nicolashefti** [comment]
@pbadenski as discussed, I modified the function implementation and also added a few test cases.

I took the liberty to add another level of "describe" to have more self-explaining test cases and not just a list assertions. I think this could be an approach we could generalize in the future. Let me know what you think.

**Comment by pbadenski** [comment]
Great - looks good to me. There's a minor issue with a corner case (`VALUE("%12") should equal 0.12`). I'll merge it anyway and we can fix this separately.

I can sway either way with regards to single "it" with list of specs vs multiple self-explaining "it blocks". Truth to be told - when I first started working with this project I was surprised it wasn't following the self-explaining style. I rather quickly got used to it and found okay to work with. There might be an extra edge in self-explaining structure as more accommodating to contributors (especially junior contributors).

P. S. I didn't know `only` trick - that's super cool. You can also do `it.only` which is awesome.

**Review by pbadenski — DISMISSED** [review]
Looks good to me.

There's a minor issue, as below test fails:
`text.VALUE('12%').should.equal(0.12);`

I think it's a small edge case, and this pull request has been in motion, so we can handle this via separate issue.

**Review by pbadenski — COMMENTED** [review]
Approved

**Review by pbadenski — APPROVED** [review]
